tachyon_core/src/lib.rs

```diff
@@ -518,7 +518,11 @@ pub struct Inserter {
 
 macro_rules! create_inserter_insert {
     ($function_name: ident, $type: ty, $value_type: expr, $value_field: ident) => {
-        pub fn $function_name(&mut self, timestamp: crate::Timestamp, value: $type) {
+        pub fn $function_name(
+            &mut self,
+            timestamp: crate::Timestamp,
+            value: $type,
+        ) -> Result<(), crate::error::TachyonErr> {
             if self.value_type != $value_type {
                 panic!("Invalid value type on insert!");
```

Comment:
A bit out of the scope of this PR, but since we are returning a Result here now, can we change this panic too?

Reply:
This should be done in a different PR, had some issues with this yesterday as the panic wasn't super clear